### PR TITLE
Improved the feedback to the user

### PR DESCRIPTION
**Pull request by stablestud**
Fixed bug were commented(#) sources were not ignored. _(This one is right here, but wrong in the commit message)_
Added failsafe to prevent the user to overwrite the original hosts file.
Added failsafe if no sources could be found.
Improved many messages. e.g. file does not exist etc.

Late at night - it would be too complicated to create for all of those things a new pull request. Sorry for that :smile:

EDIT: better form

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -66,26 +73,33 @@ case "${1}" in
...
+                      exit 1
+                    fi
                     cp /etc/hosts "${HOSTS_ORIG}"
                     # check if backup was succesfully
                     if [ ! -e "${HOSTS_ORIG}" ] ; then
                         echo "[!] Backup of /etc/hosts failed. Please backup this file manually and bypass this check by using the -f parameter."
                         exit 1
                     fi
+                else
+                  rm -f ${HOSTS_ORIG}
```
Comment (stablestud) [inline install.sh]: Controversial whether the `/etc/.hosts.original` should be deleted when the user uses the --force option or not.

File: install.sh [hunk 2]
```diff
@@ -41,23 +42,29 @@ case "${1}" in
                 fi
 
                 # check if systemd services are installed
-                if [ -e ${SYSTEMD_DIR}/adaway-linux.timer ] || [ -e ${SYSTEMD_DIR}/adaway-linux.service ] ; then
+                if [ -e "${SYSTEMD_DIR}"/adaway-linux.timer ] || [ -e "${SYSTEMD_DIR}"/adaway-linux.service ] ; then
```
Comment (sedrubal) [inline install.sh]: I'd prefer `"${SYSTEMD_DIR}/adaway-linux.timer"

File: install.sh [hunk 3]
```diff
@@ -41,23 +42,29 @@ case "${1}" in
...
 
                   echo "[!] Removing systemd service..."
                   # Unhooking the systemd service
                   systemctl stop adaway-linux.timer && systemctl disable adaway-linux.timer || echo "[!] adaway-linux.timer is missing. Have you removed it?"
                   systemctl stop adaway-linux.service && systemctl disable adaway-linux.service || echo "[!] adaway-linux.service is missing. Have you removed it?"
-                  rm ${SYSTEMD_DIR}/adaway-linux.*
+                  rm "${SYSTEMD_DIR}"/adaway-linux.*
```
Comment (sedrubal) [inline install.sh]: here the same. Please use `"` for the complete file name.

File: install.sh (hunk 1 above)
Comment (sedrubal) [reply install.sh]: I don't have a opinion on this 🤷‍

File: adaway-linux.sh [hunk 4]
```diff
@@ -52,7 +52,7 @@ mkdir -p "${TMPDIR}"
 
 # add domains from hosts-server listet in hostssources.lst
 while read src; do
-    if [[ "${src}" != "#*" ]] ; then
+    if [[ "${src}" != "#"* ]] ; then
```
Comment (sedrubal) [inline adaway-linux.sh]: What does this change do?

File: adaway-linux.sh [hunk 5]
```diff
@@ -67,49 +67,55 @@ while read src; do
...
+          >> "${TMPDIR}/hosts.downloaded"
     else
-        echo "[i] skipping $src"
+        echo "[i] Skipping ${src}"
     fi
 done < "${SCRIPT_DIR}/hostssources.lst"
 
-uniq <(sort "${TMPDIR}hosts.downloaded") > "${TMPDIR}hosts.adservers"
+# checks if any sources where downloaded
+if [ ! -e "${TMPDIR}"/hosts.downloaded ] || [ ! -s "${TMPDIR}"/hosts.downloaded ]; then
```
Comment (sedrubal) [inline adaway-linux.sh]: Please quote the complete string here, too.

File: install.sh [hunk 6]
```diff
@@ -136,7 +150,7 @@ Unit=adaway-linux.service
 [Install]
 WantedBy=timers.target
 EOL
-                        chmod u=rw,g=r,o=r ${SYSTEMD_DIR}/adaway-linux.*
+                        chmod u=rw,g=r,o=r "${SYSTEMD_DIR}"/adaway-linux.*
```
Comment (sedrubal) [inline install.sh]: and here, too 😉

File: adaway-linux.sh (hunk 4 above)
Comment (stablestud) [reply adaway-linux.sh]: The asterisks loses its function as a wildcard when it's between apostrophes.

File: install.sh (hunk 3 above)
Comment (stablestud) [reply install.sh]: @sedrubal We can't put the * into " " as it'll lose it's purpose.

File: install.sh (hunk 1 above)
Comment (stablestud) [reply install.sh]: Well then I just leave it. It'll remove the broken `.hosts.original` if it existed before.

**Pull request merged**